### PR TITLE
docs: add opa middleware docs

### LINKED ISSUE #786: Add Opa Middleware Docs
## Describe the proposal
Add basic document and examples for the new Open Policy Agent Middleware: https://github.com/dapr/dapr/issues/1617

### PR DESCRIPTION
**Pull request by RichiCoder1**
## Description

Adds a reference how-to for using the opa middleware.

## Issue reference

Closes #786

### CODE REVIEW

**Comment by RichiCoder1** [comment]
Huh, just realized that component metadata needs to be in the key value format :<

**Comment by RichiCoder1** [comment]
Updating the docs now, I wonder if I shouldn't drop the header restriction. It makes sense in Envoy since you're sending potentionally sensitive headers to an external service, but there's not good reason I can think of to exclude them by default here.

File: howto/policies-with-opa/README.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: my-policy
+  namespace: default
+spec:
+  type: middleware.http.opa
+  metadata:
+    # `includedHeaders` is a comma-seperated set of case-insensitive headers to include in the request input.
+    # By default, request headers are not passed to the policy. To use an incoming header to make an authz decision, you must include it.
```
Comment (mchmarny) [inline howto/policies-with-opa/README.md:16]: Consider: 
> Request headers are not passed to policy by default. Include to use incoming headers in authz decisions

File: howto/policies-with-opa/README.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    - name: includedHeaders
+      value: "x-my-custom-header, x-jwt-header"
+
+    # `defaultStatus` is the status code to return for denied responses if not specified in the `allow` result.
```
Comment (mchmarny) [inline howto/policies-with-opa/README.md:20]: consider:  is the status code that will be returned with denied responses

File: howto/policies-with-opa/README.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+default allow = {
+  "allow": false,
+  "status_code": 401
+}
+```
+
+### Adding Response Headers
+
+What if you you wanted to do a redirect instead? You can accomplish this using the ability to add headers to the returned result like so:
```
Comment (mchmarny) [inline howto/policies-with-opa/README.md:131]: To redirect, add headers to the returned result:

File: howto/policies-with-opa/README.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  "status_code": 301,
+  "additional_headers": {
+    "Location": "https://my.redirect.site"
+  }
+}
+```
+
+### Adding Request Headers
+
+You can also set additional headers to allowed ongoing request via the the same method:
```
Comment (mchmarny) [inline howto/policies-with-opa/README.md:147]: You can set additional headers using similar method:

File: howto/policies-with-opa/README.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  // you want to recieve via `spec.metadata.includedHeaders` (see above)
+  headers map[string]string
+  // The request scheme (e.g. http, https)
+  scheme string
+}
+```
+
+## Result
+
+The policy must set `data.http.allow` with either a `boolean` value, or an `object` value with an `allow` boolean property. A `true` `allow` will allow the request through, while a `false` value will reject the request with a `403` status. So for example, the following policy would return a `403 - Forbidden` for all requests:
```
Comment (mchmarny) [inline howto/policies-with-opa/README.md:98]: The policy must set `data.http.allow` with either a `boolean` value, or an `object` value with an `allow` boolean property. A `true` `allow` will allow the request, while a `false` value will reject the request with a `403` status. The following policy demonstrates `403 - Forbidden` result for all requests:

**Review by mchmarny — CHANGES_REQUESTED** [review]
Very detailed, thanks for putting this together. I've made a few clarifications but otherwise LGTM.

File: howto/policies-with-opa/README.md (hunk 4 above)
Comment (RichiCoder1) [reply howto/policies-with-opa/README.md:147]: I did tweak this to be terser, but I did want to differentiate redundantly between `allow -> request, denied -> response`

**Comment by RichiCoder1** [comment]
Address most all the comments, added a touch more clarity, and went ahead and added an Output reference to match the Input reference.

**Review by mchmarny — APPROVED** [review]
/lgtm

**Pull request merged**